frappe_geo_restrictions/doc_events/all.py

```diff
@@ -0,0 +1,28 @@
+import frappe
+from frappe.utils.user import is_website_user
+
+from frappe_geo_restrictions.utils.constants import ACCESS_MODES
+
+
+def before_validate(doc, method=None):
```

Comment:
Does this all handle `get_all` to any other APIs which do not call any hooks?

Reply:
By default, setting frappe.flags.readonly to True ensures no write operations are performed. before_validate, and has_permission hook overrides are just some extra check if readonly fails for any reason.